### PR TITLE
Functional tests - Add new test 'Bulk enable, disable, duplicate and delete' in product page V2'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'Bulk enable, disable, duplicate and delete' in product page V2'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/productsV2/index.js [hunk 1]
```diff
@@ -216,41 +220,80 @@ class Products extends BOBasePage {
...
-    return this.getTextContent(page, this.modalBulkdeleteProductsBody);
+    if (action === 'enable' || action === 'disable') {
+      await this.waitForSelectorAndClick(page, this.bulkActionsSelectionLink(`${action}_selection`));
+      await this.waitForVisibleSelector(page, this.modalBulkActionsProducts(`${action}_selection`));
+      textResult = await this.getTextContent(page, this.modalBulkActionsProductsBody(`${action}_selection`));
+    } else {
+      await this.waitForSelectorAndClick(page, this.bulkActionsSelectionLink(`bulk_${action}`));
+      await this.waitForVisibleSelector(page, this.modalBulkActionsProducts(`bulk_${action}`));
+      textResult = await this.getTextContent(page, this.modalBulkActionsProductsBody(`bulk_${action}`));
+    }
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/index.js]: Can't you call the method `bulkActionsProduct` ?

**Review by Progi1984 — COMMENTED** [review]
A small feedback :)

**Review by Progi1984 — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/productsV2/index.ts [hunk 2]
```diff
@@ -374,43 +383,82 @@ class Products extends BOBasePage {
...
+        1000,
+      );
+    } else {
+      await this.clickAndWaitForNavigation(page, this.modalBulkActionsProductsCloseButton(`bulk_${action}`));
+      isNotVisible = await this.elementNotVisible(
+        page,
+        this.modalBulkActionsProductsProgress(`bulk_${action}`),
+        1000,
+      );
+    }
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/index.ts]:
```suggestion
  async closeBulkActionsProgressModal(page: Page, action: string): Promise<boolean> {
    const modalBulkActionsProductsCloseButton = this.modalBulkActionsProductsCloseButton(
      (action === 'enable' || action === 'disable') ? `${action}_selection` : `bulk_${action}`,
    );
    const modalBulkActionsProductsProgress = this.modalBulkActionsProductsProgress(
      (action === 'enable' || action === 'disable') ? `${action}_selection` : `bulk_${action}`;
    );
    await this.clickAndWaitForNavigation(page, modalBulkActionsProductsCloseButton);
    return this.elementNotVisible(
      page,
      modalBulkActionsProductsProgress,
      1000,
    );
```

Comment (boubkerbribri) [reply tests/UI/pages/BO/catalog/productsV2/index.ts]: or like this 
```suggestion
  async closeBulkActionsProgressModal(page: Page, action: string): Promise<boolean> {
    let modalBulkActionsProductsCloseButton: string;
    let modalBulkActionsProductsProgress: string;
    
    if (action === 'enable' || action === 'disable') {
      modalBulkActionsProductsCloseButton = this.modalBulkActionsProductsCloseButton(
        `${action}_selection`,
      );
      modalBulkActionsProductsProgress = this.modalBulkActionsProductsProgress(
      `${action}_selection`,
      );
    } else {
    modalBulkActionsProductsCloseButton = this.modalBulkActionsProductsCloseButton(
        `bulk_${action}`,
      );
      modalBulkActionsProductsProgress = this.modalBulkActionsProductsProgress(
      `bulk_${action}`,
      );
    }
    
    await this.clickAndWaitForNavigation(page, modalBulkActionsProductsCloseButton);
    return this.elementNotVisible(
      page,
      modalBulkActionsProductsProgress,
      1000,
    );
```

File: tests/UI/pages/BO/catalog/productsV2/index.ts [hunk 3]
```diff
@@ -374,43 +383,82 @@ class Products extends BOBasePage {
...
+   * Bulk action products
    * @param page {Page} Browser tab
+   * @param action {string} Enable/disable/duplicate or delete
    * @returns {Promise<string>}
    */
-  async bulkDeleteProduct(page: Page): Promise<string> {
-    await this.waitForSelectorAndClick(page, this.modalDialogBulkDeleteButton);
-
-    await this.waitForVisibleSelector(page, this.modalBulkDeleteProductsProgressBarDone);
+  async bulkActionsProduct(page: Page, action: string): Promise<string> {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/index.ts:419]: Same maybe here

File: tests/UI/pages/BO/catalog/productsV2/index.ts [hunk 4]
```diff
@@ -374,43 +383,82 @@ class Products extends BOBasePage {
...
     await Promise.all([
       await this.waitForSelectorAndClick(page, this.productBulkMenuButton),
       await this.waitForVisibleSelector(page, this.bulkActionsDropDownMenu),
     ]);
-    await this.waitForSelectorAndClick(page, this.bulkActionsDeleteSelectionLink);
-    await this.waitForVisibleSelector(page, this.modalBulkDeleteProducts);
 
-    return this.getTextContent(page, this.modalBulkDeleteProductsBody);
+    if (action === 'enable' || action === 'disable') {
+      await this.waitForSelectorAndClick(page, this.bulkActionsSelectionLink(`${action}_selection`));
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/index.ts]: same : 
I think maybe you can have a function that return the value to put on the selector and use it on each function

Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/index.ts]: Done thx :+1:

File: tests/UI/pages/BO/catalog/productsV2/index.ts [hunk 5]
```diff
@@ -374,43 +383,68 @@ class Products extends BOBasePage {
...
    */
-  async closeBulkDeleteProgressModal(page: Page): Promise<boolean> {
-    await this.clickAndWaitForNavigation(page, this.modalBulkDeleteProductsCloseButton);
+  async closeBulkActionsProgressModal(page: Page, action: string): Promise<boolean> {
+    const modalBulkActionsProductsCloseButton = this.modalBulkActionsProductsCloseButton(
+      (action === 'enable' || action === 'disable') ? `${action}_selection` : `bulk_${action}`,
+    );
+    const modalBulkActionsProductsProgress = this.modalBulkActionsProductsProgress(
+      (action === 'enable' || action === 'disable') ? `${action}_selection` : `bulk_${action}`,
+    );
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/index.ts:444]:
```js
  getActionSelector(action: string) {
   return (action === 'enable' || action === 'disable') ? `${action}_selection` : `bulk_${action}`;
  }
  async closeBulkActionsProgressModal(page: Page, action: string): Promise<boolean> {
    const myActionSelector = await this.getActionSelector(action);
    await this.clickAndWaitForNavigation(page, this.modalBulkActionsProductsCloseButton(myActionSelector));
```

You can also have a function that can be used in the other 3 function

**Review by boubkerbribri — APPROVED** [review]
A little suggestion but LGTM

**Review by Progi1984 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh
